orchagent/switchorch.cpp

```diff
@@ -613,3 +613,22 @@ void SwitchOrch::querySwitchTpidCapability()
         set_switch_capability(fvVector);
     }
 }
+
+bool SwitchOrch::querySwitchDscpToTcCapability(sai_object_type_t sai_object, sai_attr_id_t attr_id)
+{
+    SWSS_LOG_ENTER();
+
+    /* Check if SAI is capable of handling Switch level DSCP to TC QoS map */
+    vector<FieldValueTuple> fvVector;
+    sai_status_t status = SAI_STATUS_SUCCESS;
+    sai_attr_capability_t capability;
+
+    status = sai_query_attribute_capability(gSwitchId, sai_object, attr_id, &capability);
+    if (status != SAI_STATUS_SUCCESS)
+    {
+        SWSS_LOG_WARN("Could not query switch level DSCP to TC map %d", status);
+        return false;
+    }
+
```

Comment:
You have to check `capability.set_implemented` is true as well, before returning from this function

Reply:
I didn't check for capability.set_implemented to keep changes in SAI code to minimum. Please let me know if this is still needed.